src/BinSkim.Rules/PERules/BA6004.EnableCOMDATFolding.cs

```diff
@@ -0,0 +1,101 @@
+﻿/// <summary>
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Reflection.PortableExecutable;
+
+using Microsoft.CodeAnalysis.BinaryParsers;
+using Microsoft.CodeAnalysis.BinaryParsers.PortableExecutable;
+using Microsoft.CodeAnalysis.BinaryParsers.ProgramDatabase;
+using Microsoft.CodeAnalysis.IL.Sdk;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL.Rules
+{
+    [Export(typeof(ReportingDescriptor))]
+    [Export(typeof(Skimmer<BinaryAnalyzerContext>))]
+    public class EnableCOMDATFolding : WindowsBinaryAndPdbSkimmerBase
+    {
+        /// <summary>
+        /// BA6004
+        /// </summary>
+        public override string Id => RuleIds.EnableCOMDATFolding;
+
+        /// <summary>
+        /// COMDAT folding can significantly reduce binary size by combining functions which generate identical machine code into a
+        /// single copy in the final binary.
+        /// </summary>
+
+        public override MultiformatMessageString FullDescription =>
+            new MultiformatMessageString { Text = RuleResources.BA6004_EnableCOMDATFolding_Description };
+
+        protected override IEnumerable<string> MessageResourceNames => new string[] {
+                    nameof(RuleResources.BA6004_Pass),
+                    nameof(RuleResources.BA6004_Warning_EnabledForDebug),
+                    nameof(RuleResources.BA6004_Warning_DisabledForRelease),
+                    nameof(RuleResources.NotApplicable_InvalidMetadata)
+                };
+
+        private const string AnalyzerName = RuleIds.EnableCOMDATFolding + "." + nameof(EnableCOMDATFolding);
```

Comment:
>EnableCOMDATFolding [](http://example.com/codeflow?start=52&length=19)

'EnableCommonDataFolding' or 'EnableComdatFolding' here and in all other API use, per your judgment.

Reply:
Will change